processing/generate_figure.py

```diff
@@ -532,6 +536,153 @@ def main():
         plt.savefig(fname_fig)
         logger.info('Created: '+fname_fig)
 
+        # Get T1w and T2w CSA from pandas df structure
+        if metric == "csa_t1":
+            mean_t1 = df.sort_values('vendor').values
+        elif metric == "csa_t2":
+            mean_t2 = df.sort_values('vendor').values
+
+    plt.close()
+    # Get T1w and T2w CSA per vendors and save it into 1D arrays
+    # Should be rewritten to loop or function
+    GE_mean_t1 = []
+    for f in range(1, len(mean_t1[mean_t1[:, 1] == 'GE']), 1):
+        GE_mean_t1.append(np.asarray(mean_t1[mean_t1[:, 1] == 'GE'][f, 3]))
+    GE_mean_t1 = np.concatenate(GE_mean_t1, axis=0)
+
+    Siemens_mean_t1 = []
+    for f in range(1, len(mean_t1[mean_t1[:, 1] == 'Siemens']), 1):
+        Siemens_mean_t1.append(np.asarray(mean_t1[mean_t1[:, 1] == 'Siemens'][f, 3]))
+    Siemens_mean_t1 = np.concatenate(Siemens_mean_t1, axis=0)
+
+    Philips_mean_t1 = []
+    for f in range(1, len(mean_t1[mean_t1[:, 1] == 'Philips']), 1):
+        Philips_mean_t1.append(np.asarray(mean_t1[mean_t1[:, 1] == 'Philips'][f, 3]))
+    Philips_mean_t1 = np.concatenate(Philips_mean_t1, axis=0)
```

Comment:
there is a lot of duplicated code which could be avoided by using dict, e.g. see [this](https://github.com/sct-pipeline/spine-generic/pull/52/files#diff-39e23f9cec3322e3d8bd7c7129e3541bR124)